apache/spark
# [SPARK-45148][BUILD] Upgrade scalatest related dependencies to the 3.2.17 series

### What changes were proposed in this pull request?
This pr aims upgrade `scalatest` related test dependencies to 3.2.17:

- scalatest: upgrade scalatest to 3.2.17
- scalatestplus
   - scalacheck: upgrade to `scalacheck-1-17` 3.2.17.0
   - mockito: upgrade to `mockito-4-11` to 3.2.17.0
   - selenium: uprade to `selenium-4-12` to 3.2.17.0 and `selenium-java` to 4.12.1, `htmlunit-driver` to 4.12.0, byte-buddy and byte-buddy-agent to 1.14.5

### Why are the changes needed?
The release notes as follows:

- scalatest:https://github.com/scalatest/scalatest/releases/tag/release-3.2.17
- scalatestplus
   - scalacheck-1-17: https://github.com/scalatest/scalatestplus-scalacheck/releases/tag/release-3.2.17.0-for-scalacheck-1.17
   - mockito-4-11: https://github.com/scalatest/scalatestplus-mockito/releases/tag/release-3.2.17.0-for-mockito-4.11
   - selenium-4-12: https://github.com/scalatest/scalatestplus-selenium/releases/tag/release-3.2.17.0-for-selenium-4.12


### Does this PR introduce _any_ user-facing change?
No


### How was this patch tested?

- Pass GitHub Actions
- Manual test:
   - ChromeUISeleniumSuite
   - RocksDBBackendChromeUIHistoryServerSuite

```
build/sbt -Dguava.version=32.1.2-jre -Dspark.test.webdriver.chrome.driver=/Users/yangjie01/Tools/chromedriver -Dtest.default.exclude.tags="" -Phive -Phive-thriftserver "core/testOnly org.apache.spark.ui.ChromeUISeleniumSuite"

build/sbt -Dguava.version=32.1.2-jre -Dspark.test.webdriver.chrome.driver=/Users/yangjie01/Tools/chromedriver -Dtest.default.exclude.tags="" -Phive -Phive-thriftserver "core/testOnly org.apache.spark.deploy.history.RocksDBBackendChromeUIHistoryServerSuite"
```   

```
[info] ChromeUISeleniumSuite:
[info] - SPARK-31534: text for tooltip should be escaped (1 second, 809 milliseconds)
[info] - SPARK-31882: Link URL for Stage DAGs should not depend on paged table. (604 milliseconds)
[info] - SPARK-31886: Color barrier execution mode RDD correctly (252 milliseconds)
[info] - Search text for paged tables should not be saved (1 second, 309 milliseconds)
[info] Run completed in 6 seconds, 116 milliseconds.
[info] Total number of tests run: 4
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 4, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
```

```
[info] RocksDBBackendChromeUIHistoryServerSuite:
[info] - ajax rendered relative links are prefixed with uiRoot (spark.ui.proxyBase) (1 second, 615 milliseconds)
[info] Run completed in 5 seconds, 130 milliseconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 27 s, completed 2023-9-14 11:29:27
```


### Was this patch authored or co-authored using generative AI tooling?
No
